source/agora/node/Ledger.d

```diff
@@ -108,6 +108,8 @@ public class Ledger
         // restore validator set from lastest blocks
         if (this.last_block.header.height > 0)
         {
+            UTXOSetValue[Hash] utxos =
+                this.utxo_set.getUTXOs(this.node_config.key_pair.address);
```

Comment:
This assumes the node is a Validator node. A FullNode will not have a keypair.

Reply:
Hmm. It can be a full node.

Reply:
As a workaround you could do:

```D
UTXOSetValue[Hash] utxos;
if (this.node_config.is_validator)
    utxos = this.utxo_set.getUTXOs(this.node_config.key_pair.address);
```

Reply:
Thank you. But, It must get UTXOs using a public key although it is a full node.

Reply:
I'm trying to convert the `V` of `Pair` into `address` of `KeyPair` based on what @AndrejMitrovic has given as a clue, which is adding the `opSlice` function. But it takes some more time than expected. So I'd like to have a pair programming session for that. Thanks.